servicex/servicex_adapter.py

```diff
@@ -133,11 +134,21 @@ async def get_transform_status(self, request_id: str) -> TransformStatus:
         headers = await self._get_authorization()
         retry_options = ExponentialRetry(attempts=5, start_timeout=10)
         async with RetryClient(retry_options=retry_options) as client:
-            async with client.get(url=f"{self.url}/servicex/transformation/{request_id}",
-                                  headers=headers) as r:
-                if r.status == 401:
-                    raise AuthorizationError(f"Not authorized to access serviceX at {self.url}")
-                if r.status == 404:
-                    raise ValueError(f"Transform ID {request_id} not found")
-                o = await r.json()
-                return TransformStatus(**o)
+            try:
+                async for attempt in AsyncRetrying(stop=stop_after_attempt(5),
```

Comment:
Shouldn't we add the retry for a particular error like `ClientPayloadError`(from the issue). 

AsyncRetrying() has parameter `retry` which can retry if a particular exception has occured using the `retry=retry_if_exception_type(ClientPayloadError)`

Reply:
I would put this as a 3.0.1 improvement instead, I'd rather just brute force this for now and be more subtle later.

Reply:
Shouldn't we just alter the retry statement now to exclude the `AuthorizationError` and `ValueError` exceptions with `retry_if_not_exception_type`? Seems easier than waiting.

Reply:
@gordonwatts - the keyword argument `reraise=True` of `AsyncRetrying()` enables to capture the original error. Thus no need to worry about swallowed errors.